iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -88,8 +89,13 @@ private void processPushOpens(EventProcessingRequest processingRequest) throws I
                             return;
                         }
                         request.createdAt = (int) (event.getTimestamp() / 1000.0);
-                        Response<IterableApiResponse> response = iterableService.trackPushOpen(getApiKey(processingRequest), request).execute();
-                        handleIterableResponse(response, event.getId());
+                        try {
+                            Response<IterableApiResponse> response = iterableService.trackPushOpen(getApiKey(processingRequest), request).execute();
+                            handleIterableResponse(response, event.getId());
+                        } catch (java.net.SocketTimeoutException e) {
```

Comment:
Looks like this `catch` code is duplicated multiple times.. What do you think about moving it to `handleIterableResponse`/`handleIterableListResponse`?

Reply:
That's a good callout. I was thinking about a better way to do this while updating it in 11 places...

In this example, line 93 is the one that throws the exception, so I don't think we can move the catch block into `handleIterableResponse` right?

I started looking into using a wrapper method and passing the `iterableService` method as a parameter. It looked like it was going to be a lot of work in Java. The equivalent in Python that I was thinking about would be something like:
```
def make_iterable_request(iterable_service_fn, request_body, api_key):
	try:
		return iterable_service_fn(request_body, api_key)
	except TimeoutException as e:
		raise RetriableError
```

Reply:
One way of doing this would be to pass the request to the function. Since the exception happens in `.execute()`, we could call that in the handler method and pass the request into it vs. passing the response.

Reply:
That's a good idea, I'll take a crack at it.